pkg/operators/v1/operatorgroup_types.go

```diff
@@ -45,6 +45,49 @@ type OperatorGroupSpec struct {
 	// Static tells OLM not to update the OperatorGroup's providedAPIs annotation
 	// +optional
 	StaticProvidedAPIs bool `json:"staticProvidedAPIs,omitempty"`
+
+	UpgradeStrategy *UpgradeStrategy `json:"upgradeStrategy,omitempty"`
+}
+
+// UpgradeStrategy defines the upgrade strategy for operators in the namespace.
+// There are currently two supported upgrade strategies:
+//
+// Default: OLM will only allow clusterServiceVersions to move to the replacing
+// phase from the succeeded phase. This effectively means that OLM will not
+// allow operators to move to the next version if an installation or upgrade
+// has failed.
+//
+// UnsafeFailForward: OLM will allow clusterServiceVersions to move to the
+// replacing phase from the succeeded phase or from the failed phase.
+// Additionally, OLM will generate new installPlans when a subscription references
+// a failed installPlan and the catalog has been updated with a new upgrade for
+// the existing set of operators. This upgrade strategy is unsafe and may result
+// in unexpected behavior or unrecoverable data loss unless you have deep
+// understanding of the set of operators being managed in the namespace.
+type UpgradeStrategy struct {
+	// +kubebuilder:validation:Enum=Default;UnsafeFailForward
+	// +kubebuilder:default=Default
+	Name upgradeStrategyName `json:"name"`
```

Comment:
The plan was for customers to use the exported constants, but it should be fine given the enum values defined in the CRD.

Reply:
Correct, but if I want to write a function includes the value of `Name` as a function parameter or return value, I would need to use `upgradeStrategyName`, but I can't because its unexported.

Reply:
Ah, that's fair. I'll convert it to a string.

Reply:
Rather than string, I think this should be a named, exported type.